### ISSUE TITLE
Add implicit `raises:[]` to exportc pragma or allow custom pragma

### ISSUE THREAD

**Issue by gradha** [nim-lang/Nim#845]
Right now a proc can be exported to C which silently raises an exception. Attempting to create a custom pragma including `raises:[]` doesn't seem to work, the following code compiles without problem:

``` nimrod
{.pragma: safe_export, exportc, raises: [].}

proc silent_fail(): int {.safe_export.} =
  echo "Won't know what hit him"
  result = 2
  raise newException(EInvalidValue, "Ho")
```

If the `raises:[]` can't be added to a custom pragma, it would be enough if the compiler would implicitly append that to any `exportc` pragma. The purpose of these sanity checks would be to prevent one from creating a C API which leaks nimrod exceptions impossible to handle, thus forcing you to wrap them carefully.

This implicit addition to exportc may prevent existing code from compiling, so it could be made a warning to be turned into an error on 1.0 release.

**Comment by cooldome** [comment]
It looks like a duplicate of #7216 which I have fixed in PR #7217 with a test.

can closed as fixed

**Comment by andreaferretti** [comment]
Fixed indeed
